### DISCUSSION TITLE
Why the Warning validated_self always triggers

### DISCUSSION THREAD

**Discussion by siddhartha2202** [pydantic/pydantic#10543] [Question] [answered]
Hi !!!

Since I recently updated the library the code now throws a warning. And we don't know what to do to make this Warning disappear. As you can see, even with simple classes the Warning now always throws it.

```
In [6]: from pydantic import BaseModel

In [7]: class Person(BaseModel):
   ...:     name: str
   ...:     age: int
   ...:

In [8]: p = Person(name="Person1")
---------------------------------------------------------------------------
ValidationError                           Traceback (most recent call last)
Cell In[8], line 1
----> 1 p = Person(name="Person1")

File ~/workspace/elishop/es-ventas-app/.venv/lib/python3.12/site-packages/pydantic/main.py:212, in BaseModel.__init__(self, **data)
    210 # `__tracebackhide__` tells pytest and some other tools to omit this function from tracebacks
    211 __tracebackhide__ = True
--> 212 validated_self = self.__pydantic_validator__.validate_python(data, self_instance=self)
    213 if self is not validated_self:
    214     warnings.warn(
    215         'A custom validator is returning a value other than `self`.\n'
    216         "Returning anything other than `self` from a top level model validator isn't supported when validating via `__init__`.\n"
    217         'See the `model_validator` docs (https://docs.pydantic.dev/latest/concepts/validators/#model-validators) for more details.',
    218         category=None,
    219     )

ValidationError: 1 validation error for Person
age
  Field required [type=missing, input_value={'name': 'Person1'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.9/v/missing
```

What can we do to eliminate this warning? Or what is the best practice to make it disappear?

**Comment by Viicos** [comment, marked as answer]
The warning is actually not thrown, it's just that the validation exception you get happens on line 212, and the logic to throw an (unrelated) warning happens to be defined on line 213. Your interactive session displays the source code of where the validation exception happens, together with some lines before and after.

**Reply by siddhartha2202** [reply]
Thank you very much, you are right, the problem is not that the warning is triggered, since it does not do so as you point out.

Rather, the problem is that it prints it on the screen, and now there are several scripts that are reporting this type of behavior.

Is there a way to not print this message on the screen?

For example, this is the output of one of our scripts.

```
❯ make deploy
cdk deploy --require-approval=never
/Users/xxx/workspace/xxxxx/es-ventas-app/.venv/lib/python3.12/site-packages/pydantic/main.py:214: UserWarning: A custom validator is returning a value other than self.
Returning anything other than self from a top level model validator isn't supported when validating via __init__.
See the model_validator docs (https://docs.pydantic.dev/latest/concepts/validators/#model-validators) for more details.
  warnings.warn(
#0 building with "desktop-linux" instance using docker driver

#1 [internal] load build definition from Dockerfile
#1 transferring dockerfile: 1.32kB 0.0s done
#1 DONE 0.0s
```

And in this case there is no error, but the now it prints this warning

I would greatly appreciate your help.

**Reply by Viicos** [reply]
In this case, it seems that the warning is actually triggered, so I would suggest looking into your code to see where this happens. This warnings shows up when you define _after_ model validators that don't return `self`.

If you struggle to find where the warning comes from, you can install pydantic from `main` as we recently changed the `stacklevel` of the warning to show the actual line where this happens.

**Comment by siddhartha2202** [comment]
Hi,

Thank you very much, you were right, we were able to identify that there was a class with the model_validator method and that method was not returning the `self` clause does not return. After correcting this issue it stopped sending the warning.

I really appreciate your support and the guidance you gave us.
